### PR TITLE
Error Handling

### PR DESCRIPTION
**Pull request by emanuelmetal**
# Background
We need to avoid throwing errors without the proper handling, instead add a logging system to log those errors, and in the future send them to a log service.

# Changes Done
- Added js-logger library to encapsulate the logging functionality on a basic configuration
- As per some error on develop, **husky** a pre-commit, pre-hook library was added.

### CODE REVIEW

File: webapp/app/actions/usersActions.js [hunk 1]
```diff
@@ -117,14 +113,13 @@ export function getUsers(queryParams = DEFAULT_PAGINATION_QUERY) {
...
-function handleErrors(response) {
-  if (response.statusText !== 'OK') {
-    throw Error(response.statusText);
-  }
-  return response;
+function handleErrors(error) {
+  return dispatch =>{
+    logger.error(error.message);
+    dispatch(loadingUsersFailed(error));
+  };
```
Comment (mravinale) [inline webapp/app/actions/usersActions.js]: I think we would need something global for the whole application, can we do that?

**Review by mravinale — CHANGES_REQUESTED** [review]

File: webapp/app/actions/usersActions.js (hunk 1 above)
Comment (emanuelmetal) [reply webapp/app/actions/usersActions.js]: Yes, what I think is to set up a logging service that will log (and potentially send them to a server) and leave here the dispatch of the fail, which only purpose is to update the store with the error flag to show some feedback.
@mravinale Is that what you are asking?

File: webapp/app/utils/errorService.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+      // Act
+      window.dispatchEvent( new Event('error') );
+
+      // Assert
+      expect(Logger.error).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  // describe('Set Log level', () => {
```
Comment (IvanScoles) [inline webapp/app/utils/errorService.test.js]: We should remove this describe?

**Comment by emanuelmetal** [comment]
@mravinale check this new changes please

**Review by IvanScoles — APPROVED** [review]

File: webapp/app/utils/errorService.test.js (hunk 2 above)
Comment (emanuelmetal) [reply webapp/app/utils/errorService.test.js]: Yes i'll do it

File: webapp/app/actions/usersActions.js [hunk 3]
```diff
@@ -117,14 +113,11 @@ export function getUsers(queryParams = DEFAULT_PAGINATION_QUERY) {
...
 }
 
-// TODO move to service errors utility
-function handleErrors(response) {
-  if (response.statusText !== 'OK') {
-    throw Error(response.statusText);
-  }
-  return response;
+function handleErrors(error, dispatch) {
+  errorService.logErrors('action failed', 'userActions.js');
```
Comment (mravinale) [inline webapp/app/actions/usersActions.js:121]: what if we send the error object? or the message inside the object to the error service, now all the errors are llike hardcoded

Comment (mravinale) [reply webapp/app/actions/usersActions.js:121]: proposal:
create the 3rd argument in order to pass a function to be executed.
````
errorService.logErrors(error.message 'userActions.js', ()=> somefunction());
````

having that we can remove handleErrors function and use directly the errorService

**Comment by mravinale** [comment]
@emanuelmetal what do you think implementing this as well https://medium.com/leanjs/react-16-error-handing-with-componentdidcatch-fa4e248d45d7

**Pull request merged**